ci/generate_code.bash

```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+set -Eeuo pipefail
+
+if ! python3 ci/cached-builds/gen_gha_matrix_jobs.py; then
+  # this returns nonzero when running outside of github; detect and print explanation
+  if [[ ${GITHUB_ACTIONS:-} == true ]]; then exit 1
```

Comment:
It's a pitty that the script fails when non-run in github... I consider this solution as a workaround that will be fixed and won't be needed in the near future; we may miss other reasons why the script failed when run locally with this approach.

Reply:
Yeah, it's not a good job on my part.

Reply:
I was not too had to fix, did it rn